chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -609,29 +633,33 @@ where
         }
 
         // apply undos to accounting
-        tx.outputs().iter().try_for_each(|output| match output {
+        tx.outputs().iter().for_each(|output| match output {
             TxOutput::CreateStakePool(_, _)
             | TxOutput::CreateDelegationId(_, _)
             | TxOutput::DelegateStaking(_, _) => {
-                let block_undo_fetcher = |id: Id<Block>| {
-                    self.storage
-                        .get_accounting_undo(id)
-                        .map_err(|_| ConnectTransactionError::TxVerifierStorage)
-                };
-                self.accounting_block_undo
-                    .take_tx_undo(&tx_source, &tx.get_id(), block_undo_fetcher)?
-                    .into_inner()
-                    .into_iter()
-                    .try_for_each(|undo| {
-                        self.accounting_delta_adapter.operations(tx_source).undo(undo)?;
-                        Ok(())
-                    })
+                apply_accounting_delta_undo = true;
             }
             TxOutput::Transfer(_, _)
             | TxOutput::LockThenTransfer(_, _, _)
             | TxOutput::Burn(_)
-            | TxOutput::ProduceBlockFromStake(_, _) => Ok(()),
-        })
+            | TxOutput::ProduceBlockFromStake(_, _) => { /* do nothing */ }
+        });
```

Comment:
Would this be maybe nicer with `.any()` instead of `.for_each()` and mutating? For additional benefit, it short circuits on the first `true`.